lib/segment/src/utils/scored_point_ties.rs

```diff
@@ -0,0 +1,39 @@
+use std::cmp::Ordering;
+
+use crate::types::ScoredPoint;
+
+// Newtype to provide alternative comparator for ScoredPoint which breaks ties by id
+pub struct ScoredPointTies<'a>(pub &'a ScoredPoint);
+
+impl<'a> From<&'a ScoredPoint> for ScoredPointTies<'a> {
+    fn from(scored_point: &'a ScoredPoint) -> Self {
+        ScoredPointTies(scored_point)
+    }
+}
+
+impl<'a> Ord for ScoredPointTies<'a> {
+    fn cmp(&self, other: &Self) -> Ordering {
+        let res = self.0.cmp(other.0);
+
+        // for identical scores, we fallback to sorting by ids to have a stable output
+        if res == Ordering::Equal {
+            self.0.id.cmp(&other.0.id)
+        } else {
+            res
+        }
+    }
+}
```

Comment:
We can do [`then_with`](https://doc.rust-lang.org/std/cmp/enum.Ordering.html#method.then_with):

(untested):
```suggestion
    fn cmp(&self, other: &Self) -> Ordering {
        self.0.cmp(other.0).then_with(|| self.0.id.cmp(&other.0.id))
    }
}
```